File: petshopia-APP/src/app/Model/Alojamento.ts
import { EstadoAlojamento } from "./EstadoAlojamento";

export interface Alojamento {

    alojamentoId: number; 
    animalId?: number;
    
    estadoAlojamentoId: number; 
    
    statusAlojamento: EstadoAlojamento;

}

File: petshopia-API/Model/Alojamento.cs
namespace petshopia_API.Model
{
    public class Alojamento
    {
        public Alojamento(){}
        public Alojamento(int alojamentoId, int? animalId, int estadoAlojamentoId)
        {
            this.AlojamentoId = alojamentoId;
            this.AnimalId = animalId;
            this.EstadoAlojamentoId = estadoAlojamentoId;
        }
        public int AlojamentoId { get; set; }
        public int? AnimalId{ get; set; }
        public int EstadoAlojamentoId{ get; set;}
        public EstadoAlojamento EstadoAlojamento { get; set;}
    }
}